client/src/ingredients/examples.js

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+
+function RenderSources({ sources }) {
+  return (
+    <>
+      {sources.html && <h4>HTML</h4>}
+      {/* <code> inside a <pre>?? Is that necessary? */}
+      {sources.html && (
+        <pre>
+          <code>{sources.html}</code>
+        </pre>
+      )}
+
+      {sources.css && <h4>CSS</h4>}
+      {sources.css && (
+        <pre>
+          <code>{sources.css}</code>
+        </pre>
+      )}
+
+      {sources.js && <h4>JavaScript</h4>}
+      {sources.js && (
+        <pre>
+          <code>{sources.js}</code>
+        </pre>
+      )}
+    </>
+  );
+}
+
+function RenderLiveSample({ example }) {
+  const srcdoc = `<html>
+  <head>
+      <meta charset="utf-8">
+      <style type="text/css">${example.sources.css}</style>
+      <title>${example.description.title}</title>
+  </head>
+  <body>${example.sources.html}
+      <script>${example.sources.js || ""}</script>
+  </body>
+</html>`;
+  return (
+    <>
+      <h4>Result</h4>
+      <iframe
+        className="live-sample-frame sample-code-frame"
+        srcDoc={srcdoc}
+        title={example.description.title}
+        id="frame_Live_example"
+        width={example.description.width}
+        height={example.description.height}
+        frameBorder={0}
+      >
+        >
+      </iframe>
+    </>
+  );
+}
+
+function RenderExample({ example }) {
+  return (
+    <>
+      {example.description.title && <h3>{example.description.title}</h3>}
+
+      {example.description.title && (
+        <div
+          dangerouslySetInnerHTML={{ __html: example.description.content }}
+        />
+      )}
+
+      <RenderSources sources={example.sources} />
+
+      {/* XXX this is an odd condition */}
+      {example.description.width && <RenderLiveSample example={example} />}
+    </>
+  );
+}
+
+export function Examples({ document }) {
+  return (
+    <>
+      <h2>Examples</h2>
+      {document.examples.map((example, i) => (
+        <RenderExample key={example.description.title + i} example={example} />
```

Comment:
If there are multiple examples, there could, in theory be multiple iframes and what's stopping it from getting the same ID more than once? 
So, even if we do `id={this.slugifyTitle(example.description.title)}` it could repeat if the titles aren't distinct.

Reply:
Yeah, so we should probably validate that slugified example titles are unique.

Reply:
-> https://github.com/mdn/stumptown-experiment/issues/47